guests-selector/index.js

```diff
@@ -0,0 +1,98 @@
+        this.maxAdults = parseInt(this.getAttribute('max-adults')) || 5;
+        this.maxChildren = parseInt(this.getAttribute('max-children')) || 5;
+        this.maxChildAge = parseInt(this.getAttribute('max-child-age')) || 17;
+        this.label = this.getAttribute('label') || 'Room';
+
+        this.state = {
+            adults: 0,
+            children: 0,
+        };
+    }
+
+    connectedCallback() {
+        this.render();
+        this.updateDisplay();
+        this.attachEventListeners();
+    }
+
+    render() {
+        this.innerHTML = `
+            <div class="room-container">
+                <div class="room-header">
+                    <span>${this.label} (max ${this.maxTotalGuests} guests)</span>
+                    <button type="button" class="remove-room">Remove</button>
+                </div>
+                <div class="guest-controls">
+                    <number-picker label="Adults" min="0" max="${this.maxAdults}" initialValue="0"></number-picker>
+                    <number-picker label="Children" min="0" max="${this.maxChildren}" initialValue="0"></number-picker>
+                    <div class="children-age-dropdown" id="children-age-container"></div>
+                </div>
+            </div>
+        `;
+    }
+
+    attachEventListeners() {
+        // Listen for the valueChanged event from the "Adults" picker
+        this.querySelector('number-picker[label="Adults"]').addEventListener('valueChanged', (e) => {
+            this.state.adults = e.detail.value;
+            this.updateTotalGuests();
+            this.updateDisplay(); // Call updateDisplay to reflect changes
+        });
+
+        // Listen for the valueChanged event from the "Children" picker
+        this.querySelector('number-picker[label="Children"]').addEventListener('valueChanged', (e) => {
+            this.state.children = e.detail.value;
+            this.updateTotalGuests();
+            this.renderChildrenAgeDropdown();
+            this.updateDisplay(); // Call updateDisplay to reflect changes
+        });
+
+        this.querySelector('.remove-room').addEventListener('click', () => this.removeRoom());
+    }
+
+    updateTotalGuests() {
+        const totalGuests = this.state.adults + this.state.children;
+        if (totalGuests > this.maxTotalGuests) {
+            alert(`Max total guests (${this.maxTotalGuests}) exceeded!`);
+        }
+    }
+
+    updateDisplay() {
+        // Updates the displayed values for adults and children
+        this.querySelector('number-picker[label="Adults"] .number-value').textContent = this.state.adults;
+        this.querySelector('number-picker[label="Children"] .number-value').textContent = this.state.children;
+    }
+
+    renderChildrenAgeDropdown() {
+        const container = this.querySelector('#children-age-container');
+        container.innerHTML = ''; // Clear previous dropdowns
+
+        for (let i = 0; i < this.state.children; i++) {
```

Comment:
@samuelsilvapt I wonder if template literal will make this easier to read and mean we can use less for loops?

Also I think the label value will need to be an attribute value so that it can be change by the user or at least different per website that uses it. Also need a aria label or arialabledby value on the select